### ISSUE TITLE
gdscript-godot not autoloading and keymap is obsolete

### ISSUE THREAD

**Issue by chirimantecman** [GDQuest/emacs-gdscript-mode#59]
**I'm submitting a...**

- [x] Bug report.
- [ ] Feature request.

## Bug report ##

**What is the current behavior?**
1. When gdscript-mode is called, gdscript-godot is not loaded, thus there is no access to functions like gdscript-godot-run-project and similar.  When calling one of these functions the following error is shown in the minibuffer:

`Wrong type argument: commandp, gdscript-godot-run-project`

(replace with the actual function called)

2. Also, in gdscript-mode.el, variable gdscript-mode-map has key definitions that point to apparently older names for the functions mentioned in the previous paragraph.

**What is the expected behavior?**
1. gdscript-godot should be loaded when gdscript-mode is called.
2. in gdscript-mode.el, the key definitions in the keymap should correspond to the correct function names.

**Tell us the steps to reproduce the bug, and if possible share a minimal demo of the problem.** 

1. Open a .gd file.
2. Do M-x gdscript-godot-run-project.
3. Do C-c C-p.

## Other ##
gdscript-mode version: 1.1.0
Emacs version: 26.3

The attached gdscript-mode.el file fixed both issues for me.

[gdscript-mode.txt](https://github.com/GDQuest/emacs-gdscript-mode/files/4691036/gdscript-mode.txt)

**Comment by NathanLovato** [comment]
I probably broke something with my last commit. It's working here but that might be Doom Emacs's caching at play. I'll get to it ASAP. Thanks for the report

**Comment by NathanLovato** [comment]
Are you using vanilla emacs or a distribution like Spacemacs or Doom? Did you have an older version of the add-on installed, with byte-compiled files?

gdscript-mode works fine for me, and the gdscript-docs file is not causing problems. I can't reproduce your issue right now.

**Comment by chirimantecman** [comment]
Hi,

I'm using vanilla Emacs, version 26.3, and I'm using gdscript-mode 1.1.0,
from MELPA-stable.

KR,
Cristian

**Comment by NathanLovato** [comment]
Even with vanilla, it's loading without problem for me, and I can run the gdscript-docs-* commands.

**Comment by VlachJosef** [comment]
I can see same error when running 1.1.0 from Melpa stable. I'm not sure if it as option but can't you just use Melpa (non stable)?

**Comment by chirimantecman** [comment]
@NathanLovato: just to make sure there's no confusion, I'm not talking about the gdscript-docs-* functionality, that isn't in version 1.1.0.  I see that in the new version already fixes this (in gdscript-mode.el).

@VlachJosef: Yeah, I can probably use the new version, no problem.  Although es per my original post,I already fixed this particular issue in the one I have installed.  With the code I attached.

Just wanted to announce this error in version 1.1.0 for other to know in case they use that one too.
